File: src/endpoints/article/put_item.py
from src import output
from src.libs.db import DB, Table
from src.modules.verify import checking_token
from .__libs__ import Status

async def put_item(req = None, _db: DB = None, _check_token = True, **kwargs):

    # set values
    result = None
    db = _db if _db else DB().connect()

    try:
        # checking token
        if _check_token: checking_token(req, db)

        # check ready mode item
        item = db.get_item(
            table_name = Table.ARTICLE.value,
            fields=[ 'srl' ],
            where=[ f'mode LIKE \'{Status.READY}\'' ],
        )
        if item:
            data = item.get('srl')
        else:
            data = db.add_item(
                table_name=Table.ARTICLE.value,
                placeholders=[ { 'key': 'mode', 'value': ':mode' } ],
                values={ 'mode': Status.READY },
            )

        # set result
        result = output.success({
            'message': 'Complete add article.',
            'data': data,
        }, _req=req)
    except Exception as e:
        result = output.exc(e, _req=req)
    finally:
        if not _db and db: db.disconnect()
        return result

File: src/routes/article/put-item.ts
import DB, { db } from '@/classes/DB'
import ServiceError from '@/classes/ServiceError'
import { parseJSON } from '@/libs/objects'
import * as helper from './__helper'

export default async function putItem()
{
  try
  {
    // get ready data
    let article = db.getData({
      table: DB.TABLE.ARTICLE,
      where: `mode LIKE \'${helper.STATUS.READY}\'`
    })

    // 대기 데이터가 없으면 데이터를 만들고 가져온다.
    if (!article.data)
    {
      const newArticle = db.addData({
        table: DB.TABLE.ARTICLE,
        values: [
          { key: 'mode', value: helper.STATUS.READY },
        ],
      })
      article = db.getData({
        table: DB.TABLE.ARTICLE,
        where: `srl = ${newArticle.data}`,
      })
    }

    // filtering article data
    if (article.data?.json)
    {
      article.data.json = parseJSON(article.data.json)
    }

    return article.data
  }
  catch (_e: any)
  {
    throw new ServiceError('Failed to add Article.', {
      status: _e.status,
      text: _e.message,
      cause: _e,
    })
  }
}
